scipy/_lib/_array_api.py

```diff
@@ -283,13 +283,20 @@ def xp_assert_equal(actual, desired, check_namespace=True, check_dtype=True,
     return np.testing.assert_array_equal(actual, desired, err_msg=err_msg)
 
 
-def xp_assert_close(actual, desired, rtol=1e-07, atol=0, check_namespace=True,
+def xp_assert_close(actual, desired, rtol=None, atol=0, check_namespace=True,
                     check_dtype=True, check_shape=True, err_msg='', xp=None):
     __tracebackhide__ = True  # Hide traceback for py.test
     if xp is None:
         xp = array_namespace(actual)
     desired = _strict_check(actual, desired, xp, check_namespace=check_namespace,
                             check_dtype=check_dtype, check_shape=check_shape)
+
+    floating = xp.isdtype(desired.dtype, ('real floating', 'complex floating'))
+    if rtol is None and floating:
+        rtol = xp.finfo(desired.dtype).eps**0.5 * 4
```

Comment:
Maybe helpful to add a quick comment here: is there any special about 4 or just experimentally worked well?

Reply:
Good question. It is a nice number that puts us about halfway between `sqrt(eps)` and the old `1e-7` default for floats, and it was on my mind because it's used in the definition of tolerances in `optimize._zeros.py`. And yeah, it happened to be enough so that no old tests started failing. I can add a comment that any multiple that keeps us under the old default would be OK. `sqrt(eps)` can be theoretically motivated in some contexts.

Reply:
Addressed in gh-20625 so I'll take the liberty of resolving.